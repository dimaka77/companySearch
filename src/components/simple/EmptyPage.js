import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import { ImageEmptyState, ImageNoResults } from './';

const style = {
	container: {
		textAlign: 'center',
		width: '100%'
	},
	image: {
		marginLeft: '45px',
		width: '320px'
	},
	text: {
		color: 'rgba(74,74,74,.22)',
		fontFamily: 'montserrat_light, sans-serif',
		fontSize: '32px',
		margin: '30px 0 40px'
	}
};

const EmptyPage = (props) => {
	const { text, initialLoad, classes } = props;
	const components = {
		empty: ImageEmptyState,
		noResults: ImageNoResults
	};
	const View = initialLoad ? components.empty : components.noResults;

	return (
		<div style={style.container}>
			<Typography align="center" classes={{ root: classes.text }}>{text}</Typography>
			<View style={style.image} />
		</div>
	);
};

EmptyPage.defaultProps = {
	classes: {},
	text: '',
	initialLoad: true
};

EmptyPage.propTypes = {
	classes: PropTypes.object,
	text: PropTypes.string,
	initialLoad: PropTypes.bool
};

export default withStyles(style)(EmptyPage);
