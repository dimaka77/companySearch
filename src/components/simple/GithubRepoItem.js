import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Avatar from '@material-ui/core/Avatar';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Chip from '@material-ui/core/Chip';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
    avatar: {
        display: 'inline-block',
        margin: 10,
        width: 60,
        height: 60,
    },
    avatarImg: {
        objectFit: 'fill'
    },
    tabSection: {
        padding: '0 15px',
        position: 'relative',
        top: '60px',
        fontSize: '14px',
        fontWeight: 500
    },
});
const GithubRepoItem = (props) => {
    const {
        classes,
        open_issues_count,
        src,
        watchers,
        stargazers_count,
        htmlUrl
    } = props;

    return (
        <ListItem>
            <Avatar
                alt="github-image"
                classes={{ root: classes.avatar, img: classes.avatarImg }}
                src={src}
            />
            <ListItemText>
                <div>
                    <a
                        href={htmlUrl}
                        target="_blank"
                        rel="noopener noreferrer"
                    >
                        {htmlUrl}
                    </a>
                </div>
                <Fragment>
                    <Chip
                        label={`watchers: ${watchers}`}
                        className={classes.chip}
                    />
                    <Chip
                        label={`stargazers: ${stargazers_count}`}
                        className={classes.chip}
                    />
                    <Chip
                        label={`open issues: ${open_issues_count}`}
                        className={classes.chip}
                    />
                </Fragment>
            </ListItemText>
        </ListItem>
    )
}
GithubRepoItem.defaultProps = {
    classes: {},
    open_issues_count: 0,
    src: '',
    watchers: 0,
    stargazers_count: 0,
    htmlUrl: ''
};

GithubRepoItem.propTypes = {
    classes: PropTypes.object,
    open_issues_count: PropTypes.number,
    src: PropTypes.string,
    watchers: PropTypes.number,
    stargazers_count: PropTypes.number,
    htmlUrl: PropTypes.string
};

export default withStyles(styles)(GithubRepoItem);
