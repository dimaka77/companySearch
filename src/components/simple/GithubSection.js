


import React from 'react';
import PropTypes from 'prop-types';
import Avatar from '@material-ui/core/Avatar';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
    avatar: {
        display: 'inline-block',
        margin: 10,
        width: 60,
        height: 60,
    },
    avatarImg: {
        objectFit: 'fill'
    },
    tabSection: {
        padding: '0 15px',
        position: 'relative',
        top: '60px',
        fontSize: '14px',
        fontWeight: 500
    },
});
const GithubSection = (props) => {
    const {
        classes,
        key,
        src,
        description,
        email,
        htmlUrl
    } = props;
    return (
        <section
            className={classes.tabSection}
            key={key}
        >
            <div>
                <Avatar
                    alt="github-image"
                    classes={{ root: classes.avatar, img: classes.avatarImg }}
                    src={src}
                />
            </div>
            <div>{`Description: ${description}`}</div>
            <div>{`Email: ${email}`}</div>
            <div>URL:&nbsp;
					<a
                    href={htmlUrl}
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    {htmlUrl}
                </a>
            </div>
        </section>
    )
}
GithubSection.defaultProps = {
    classes: {},
    key: '0',
    src: '',
    description: '',
    email: '',
    htmlUrl: ''
};

GithubSection.propTypes = {
    classes: PropTypes.object,
    key: PropTypes.string,
    src: PropTypes.string,
    description: PropTypes.string,
    email: PropTypes.string,
    htmlUrl: PropTypes.string
};

export default withStyles(styles)(GithubSection);
