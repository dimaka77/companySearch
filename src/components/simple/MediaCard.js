


import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';

const styles = theme => ({
    avatar: {
        display: 'inline-block',
        margin: 10,
        width: 60,
        height: 60,
    },
    avatarImg: {
        objectFit: 'fill'
    },
    card: {
        display: 'flex'
    },
    details: {
        display: 'flex',
        flexDirection: 'column'
    },
    divider: {
        width: 1,
        height: 28,
        margin: 4
    },
    cardTextBlock: {
        display: 'inline-block',
        height: '75px',
        verticalAlign: 'top',
        padding: '10px 15px'
    }
});
const MediaCard = (props) => {
    const {
        classes,
        src,
        title,
        description
    } = props;
    return (
        <Card className={classes.card}>
            <div className={classes.details}>
                <CardContent>
                    <Avatar
                        alt="company-image"
                        classes={{ root: classes.avatar, img: classes.avatarImg }}
                        src={src}
                    />
                    <div className={classes.cardTextBlock}>
                        <Typography component="h5" variant="h5">
                            {title}
                        </Typography>
                        <Typography variant="subtitle1" color="textSecondary">
                            {description}
                        </Typography>
                    </div>
                </CardContent>
            </div>
        </Card>
    )
}
MediaCard.defaultProps = {
    classes: {},
    src: '',
    title: '',
    description: ''
};

MediaCard.propTypes = {
    classes: PropTypes.object,
    src: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string
};

export default withStyles(styles)(MediaCard);
