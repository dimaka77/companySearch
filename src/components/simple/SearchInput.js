import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import SearchIcon from '@material-ui/icons/Search';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';

const styles = (error = false) => ({
    inputRoot: {
		padding: '2px 4px',
		display: 'flex',
		alignItems: 'center',
		width: 400,
        margin: '0 auto',
        ...error && {
            border: '1px solid red'
        }
    },
    input: {
        marginLeft: 8,
        flex: 1
    },
    divider: {
        width: 1,
        height: 28,
        margin: 4
    },
    error: {
        padding: '2px 4px',
        width: 400,
        margin: '0 auto',
        color: 'red'
    }
});
const SearchInput = (props) => {
    const {
        classes,
        error,
        handleInputChange,
        handleSearchClick,
        value,
        placeholder
    } = props;
    return (
        <Fragment>
        <Paper style={styles(error).inputRoot} elevation={4}>
            <InputBase
                autoFocus
                className={classes.input}
                onChange={handleInputChange}
                value={value}
                placeholder={placeholder}
            />
            <Divider className={classes.divider} />
            <IconButton
                className={classes.iconButton}
                aria-label="Search"
                onClick={handleSearchClick}
            >
                <SearchIcon />
            </IconButton>
        </Paper>
            {error && <Typography style={styles().error}>Please enter company name</Typography> }
        </Fragment>
    )
}
SearchInput.defaultProps = {
    placeholder: "Enter company name...",
    classes: {},
    error: false,
    handleInputChange: () => {},
    handleSearchClick: () => {},
    value: ''
};

SearchInput.propTypes = {
    placeholder: PropTypes.string,
    classes: PropTypes.object,
    error: PropTypes.bool,
    handleInputChange: PropTypes.func,
    handleSearchClick: PropTypes.func,
    value: PropTypes.string
};

export default withStyles(styles)(SearchInput);
