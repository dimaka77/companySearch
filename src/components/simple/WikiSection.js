import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

const styles = () => ({
    tabSection: {
        padding: '0 15px',
        position: 'relative',
        top: '60px',
        fontSize: '14px',
        fontWeight: 500
    }
});
const WikiSection = (props) => {
    const {
        classes,
        key,
        html
    } = props;
    return (
        <section
            className={classes.tabSection}
            key={key}
        >
            <div dangerouslySetInnerHTML={{ __html: html }} />
        </section>
    );
}
WikiSection.defaultProps = {
    classes: {},
    key: 0,
    html: ''
};

WikiSection.propTypes = {
    classes: PropTypes.object,
    key: PropTypes.number,
    html: PropTypes.string
};

export default withStyles(styles)(WikiSection);
