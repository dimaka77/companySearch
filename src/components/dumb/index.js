import EmptyPage from './EmptyPage';
import ImageEmptyState from './ImageEmptyState';
import ImageNoResults from './ImageNoResults';

export {
    EmptyPage,
    ImageEmptyState,
    ImageNoResults
};
