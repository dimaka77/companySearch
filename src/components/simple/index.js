import EmptyPage from './EmptyPage';
import ImageEmptyState from './ImageEmptyState';
import ImageNoResults from './ImageNoResults';
import SearchInput from './SearchInput';
import MediaCard from './MediaCard';
import WikiSection from './WikiSection';
import GithubSection from './GithubSection';
import GithubRepoItem from './GithubRepoItem';

export {
    EmptyPage,
    ImageEmptyState,
    ImageNoResults,
    SearchInput,
    MediaCard,
    WikiSection,
    GithubSection,
    GithubRepoItem
};
