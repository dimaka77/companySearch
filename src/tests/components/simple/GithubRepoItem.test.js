import React from 'react';
import renderer from 'react-test-renderer';
import { GithubRepoItem } from '../../../components/simple';

it('renders GithubRepoItem component correctly', () => {
    const tree = renderer
        .create(<GithubRepoItem />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
