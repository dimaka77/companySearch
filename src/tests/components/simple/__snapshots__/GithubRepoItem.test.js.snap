// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GithubRepoItem component correctly 1`] = `
<li
  className="MuiListItem-root-4 MuiListItem-default-7 MuiListItem-gutters-12"
  disabled={false}
>
  <div
    className="MuiAvatar-root-16 GithubRepoItem-avatar-1 MuiAvatar-colorDefault-17"
  />
  <div
    className="MuiListItemText-root-19"
  >
    <span
      className="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-primary-22"
    >
      <div>
        <a
          href=""
          rel="noopener noreferrer"
          target="_blank"
        >
          
        </a>
      </div>
      <div
        className="MuiChip-root-61"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiChip-label-80"
        >
          watchers: 0
        </span>
      </div>
      <div
        className="MuiChip-root-61"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiChip-label-80"
        >
          stargazers: 0
        </span>
      </div>
      <div
        className="MuiChip-root-61"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiChip-label-80"
        >
          open issues: 0
        </span>
      </div>
    </span>
  </div>
</li>
`;
